source/common/http/conn_manager_impl.cc

```diff
@@ -173,6 +173,9 @@ StreamDecoder& ConnectionManagerImpl::newStream(StreamEncoder& response_encoder)
   new_stream->response_encoder_ = &response_encoder;
   new_stream->response_encoder_->getStream().addCallbacks(*new_stream);
   config_.filterFactory().createFilterChain(*new_stream);
+  if (underlying_connection_above_high_watermark_) {
```

Comment:
There's 2 ways things can be delayed

You can have the connection get blocked when you have a stream but no upstream connection.
At this point you get informed when the router registers.  But you can also have the connection get blocked before the stream is *created*.  This handles that case.  We tell the stream it's blocked and it remembers to tell the router later.